internal/profile/install_profile.go

```diff
@@ -0,0 +1,212 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package profile
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/elastic-package/internal/configuration/locations"
+)
+
+// CreateProfile installs a new profile at the given package path location.
+// overwriteExisting determines the behavior if a profile with the given name already exists.
+// On true, it'll overwrite the profile, on false, it'll copy the existing profile over to profileName_old
+func CreateProfile(elasticPackagePath string, profileName string, overwriteExisting bool) error {
+
+	profile, err := NewConfigProfile(elasticPackagePath, profileName)
+	if err != nil {
+		return errors.Wrap(err, "error creating profile")
+	}
+
+	//check to see if we have an existing profile at that location.
+	exists, err := profile.profileAlreadyExists()
+	if err != nil {
+		return errors.Wrap(err, "error checking for existing profile")
+	}
+	if exists {
+		localChanges, err := profile.localFilesChanged()
+		if err != nil {
+			return errors.Wrapf(err, "error checking for changes in %s", profile.ProfilePath)
+		}
+
+		// If there's changes and we've selected CreateNew, move the old path
+		// TODO: do we want this to pe appended with some kind of version string instead?
+		if localChanges && !overwriteExisting {
+			if localChanges && profileName == DefaultProfile {
+				fmt.Printf("WARNING: default profile has been changed by user or updated by elastic-package. The current profile will be moved to default_old.\n")
```

Comment:
Since we're using `WARNING` here this looks like a good candidate for a logging statement. See https://github.com/elastic/elastic-package/blob/cb436d9f36d9c0b675184217df0c643f46064125/internal/logger/logger.go#L52-L55